### ISSUE TITLE
CPU-dependent Test failures with PGI compilers

### ISSUE THREAD

**Issue by drroe** [Amber-MD/cpptraj#841]
Some original discussion was in #840 

Parts or all of the following tests fail when compiled with PGI compilers, but only for certain types of hardware:
```
Test_Cluster_RandomSieve/Test_Error.dat
Test_Cluster_Sieve/Test_Error.dat
Test_Cluster_SymmRMSD/Test_Error.dat
Test_CurveFit/Test_Error.dat
Test_EvalPlateau/Test_Error.dat
Test_GIST/Test_Error.dat
```

Failures occur on the following platforms:
- Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz

Failures do not occur on the following platforms:
- Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz

It's possible the failures are optimization-related; e.g. the Intel 6130 supports AVX-512 while the E5-2670 does not. I will try compiling with no optimizations and see if the test failures go away.

**Comment by drroe** [comment]
Even with `-O0` the tests still fail.

**Comment by drroe** [comment]
Using flags recommended by [Blue Waters](https://bluewaters.ncsa.illinois.edu/pgi-compiler):
```
-m64 -mcmodel=medium -Mdalign -Mllalign -Munroll -Kieee -O2 -fastsse -Mipa=fast
```
results in the same test failures, plus `Test_CCP4/Test_Error.dat`.

**Comment by drroe** [comment]
Finally addressed by #876 !
